IdentityCore/src/cache/token/MSIDCredentialCacheItem.m

```diff
@@ -329,20 +337,24 @@ - (BOOL)matchesWithRealm:(nullable NSString *)realm
         if ((clientId && [self.clientId.msidNormalizedString isEqualToString:clientId.msidNormalizedString])
             || (familyId && [self.familyId.msidNormalizedString isEqualToString:familyId.msidNormalizedString]))
         {
+            MSID_LOG_WITH_CTX(MSIDLogLevelVerbose, nil, @"(MSIDCredentialCacheItem) cached item supserset match; actual client ID: %@, expected client ID: %@, actual family ID: %@, expected family ID: %@", self.clientId.msidNormalizedString, clientId.msidNormalizedString, self.familyId.msidNormalizedString, familyId.msidNormalizedString);
             return YES;       
         }
-
+        
+        MSID_LOG_WITH_CTX(MSIDLogLevelVerbose, nil, @"(MSIDCredentialCacheItem) cached item superset mismatch; cached item had both invalid clientID and familyID. actual client ID: %@, expected client ID: %@, actual family ID: %@, expected family ID: %@", self.clientId.msidNormalizedString, clientId.msidNormalizedString, self.familyId.msidNormalizedString, familyId.msidNormalizedString);
         return NO;
     }
     else
     {
         if (clientId && ![self.clientId.msidNormalizedString isEqualToString:clientId.msidNormalizedString])
         {
+            MSID_LOG_WITH_CTX(MSIDLogLevelVerbose, nil, @"(MSIDCredentialCacheItem) cached item clientID mismatch; actual client ID: %@, expected client ID: %@", self.clientId.msidNormalizedString, clientId.msidNormalizedString);
             return NO;
         }
 
         if (familyId && ![self.familyId.msidNormalizedString isEqualToString:familyId.msidNormalizedString])
         {
+            MSID_LOG_WITH_CTX(MSIDLogLevelVerbose, nil, @"(MSIDCredentialCacheItem) cached item familyID mismatch; actual family ID: %@, expected family ID: %@", self.familyId.msidNormalizedString, familyId.msidNormalizedString);
```

Comment:
Do you need to perform `msidNormalizedString` for logging? It's an expensive operation, so maybe we don't do it for logs?

Reply:
sounds good

Reply:
In hindsight, I think it would suffice to just print out that they mismatched, instead of actually printing out the values, since PII masking will turn them into null not-null anyways.